### PR TITLE
Homework part 3: Rspec Testing

### PR DESCRIPTION
**Pull request by SOUVIGHO**
Implemented Rspec in Backed HW assignment
Added 1 example/test case in model spec
Added 4 example/test case in lib spec
Added 10 example/test case in controller spec

### CODE REVIEW

File: spec/lib/omdbapi_spec.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+include OmdbApi
+
+RSpec.describe OmdbApi, type: :lib do
+
+  let! (:response) do
```
Comment (oracle2g) [inline spec/lib/omdbapi_spec.rb:6]: no space 
```ruby let!(:response)```

Comment (oracle2g) [inline spec/lib/omdbapi_spec.rb:6]: use https://www.betterspecs.org/#contexts

File: spec/models/movie_spec.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require 'rails_helper'
+
+RSpec.describe Movie, type: :model do
+  let! (:movies) do
+    [Movie.create ,Movie.create ,Movie.create]
+  end
+
+  it "checks if create Movie is working" do
```
Comment (oracle2g) [inline spec/models/movie_spec.rb:8]: Don't think this spec is necessary, more relevant ones would be to test 
1. Assert on Column keys returned 
2. Happy path testing and edge cases

File: spec/requests/movies_spec.rb [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    end
+    it "returns http success" do
+      expect(response).to have_http_status(:success)
+    end
+    it "returns JSON response" do
+      expect(response.content_type).to match(a_string_including('application/json'))
+    end
+  end
+  describe "POST /movies" do
+    let! (:valid_param) do
```
Comment (oracle2g) [inline spec/requests/movies_spec.rb:16]: no space

File: spec/requests/movies_spec.rb [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+      {
+        title: "Test Movie 41",
+        poster: "https:\\example.com.jpg",
+        rating: "7.8",
+        graphenabled: true
+      }
+    end
+    it "returns http success" do
```
Comment (oracle2g) [inline spec/requests/movies_spec.rb:24]: Add test case when failure occurs

File: spec/requests/movies_spec.rb [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+    it "returns http success" do
+      delete "/movies/"+newid.to_s
+      expect(response).to have_http_status(:success)
+    end
+    it "deletes one new record in movies collection" do
+      expect { delete "/movies/"+newid.to_s }.to change(Movie, :count).by(-1)
+    end
+  end
+end
```
Comment (oracle2g) [inline spec/requests/movies_spec.rb:73]: nit: insert newline

File: spec/lib/omdbapi_spec.rb [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  end
+
+  it "checks if the rating is valid" do
+    expect(response["imdbRating"].to_f).to be_a_kind_of(Float) & (be <= 10)
+  end
+  
+  it "checks if the poster url is valid" do
+    expect(response["Poster"]).to start_with("https:") & end_with(".jpg")
+  end
+
```
Comment (oracle2g) [inline spec/lib/omdbapi_spec.rb:24]: nit: extra space

**Review by oracle2g — CHANGES_REQUESTED** [review]
Initial review complete,

Please add https://github.com/rubocop/rubocop to your gemfile and run ```ruby bundle exec rubocop```

File: spec/lib/omdbapi_spec.rb (hunk 1 above)
Comment (SOUVIGHO) [reply spec/lib/omdbapi_spec.rb:6]: Removed Spec

Comment (SOUVIGHO) [reply spec/lib/omdbapi_spec.rb:6]: Used context to make the tests clear

File: spec/lib/omdbapi_spec.rb (hunk 6 above)
Comment (SOUVIGHO) [reply spec/lib/omdbapi_spec.rb:24]: Where should I add extra space?

File: spec/models/movie_spec.rb (hunk 2 above)
Comment (SOUVIGHO) [reply spec/models/movie_spec.rb]: Added Test Case on column keys.
Added Happy Path test case.

File: spec/requests/movies_spec.rb (hunk 3 above)
Comment (SOUVIGHO) [reply spec/requests/movies_spec.rb]: Removed extra space and added context

File: spec/requests/movies_spec.rb (hunk 4 above)
Comment (SOUVIGHO) [reply spec/requests/movies_spec.rb]: Added test case to raise error
